app/src/main/java/com/vestrel00/daggerbutterknifemvp/ui/common/presenter/Presenter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017 Vandolf Estrellado
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.vestrel00.daggerbutterknifemvp.ui.common.presenter;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+
+/**
+ * A presenter that defines its own lifecycle methods.
+ */
+public interface Presenter {
+
+    /**
+     * Starts the presentation. This should be called in the view's (Activity or Fragment)
+     * onCreate() or onViewStatedRestored() method respectively.
+     *
+     * @param savedInstanceState the saved instance state that contains state saved in
+     *                           {@link #onSaveInstanceState(Bundle)}
+     */
+    void onStart(@Nullable Bundle savedInstanceState);
```

Comment:
Another example demonstrates this save / restore state process: https://github.com/vestrel00/android-dagger-butterknife-mvp/issues/41